### ISSUE TITLE
Headers accept non-ASCII content

### ISSUE THREAD

**Issue by singpolyma** [purebred-mua/purebred-email#41]
set (MIME.headers . at (s"Subject")) (encodeUtf8 "<some unicode>")

This works, but rendering the resulting message gives me, not the encoded words I expected, but raw unsafe 8bit content.

**Comment by frasertweedale** [comment]
@singpolyma do you mean:
```haskell
set (MIME.headers . at (s"Subject")) (Just $ encodeUtf8 "<some unicode>")
                                   -- ^^^^
```

Anyhow I think we need to document this better for now, and going forward perhaps use a "7bit" newtype wrapper with smart constructor at this "first level" of header processing, that will make it impossible to write 8-bit data into headers.

Thoughts?

**Comment by singpolyma** [comment]
Yeah, maybe want to use https://hackage.haskell.org/package/ascii in the right places -- though if a higher-level API existed that did the "right thing" then maybe it would be less likely than anyone would try this and less of a big deal.

**Comment by frasertweedale** [comment]
Fixed in fa4f36e01b6f4bd6e950f378c0feeee713dcd640.
